stackslib/src/net/api/postblock_proposal.rs

```diff
@@ -182,16 +171,28 @@ impl NakamotoBlockProposal {
         let mainnet = self.chain_id == CHAIN_ID_MAINNET;
         if self.chain_id != chainstate.chain_id || mainnet != chainstate.mainnet {
             return Err(BlockValidateReject {
+                block: self.block.clone(),
```

Comment:
I think it would simplify this function a bit to just have a new type:

```rust
#[derive(Debug, Clone, PartialEq)]
pub struct BlockValidateRejectReason {
    pub reason: String,
    pub reason_code: ValidateRejectCode,
}
```

and change the return type of `validate` to `Result<BlockValidateOk, BlockValidateReject>`

with the previous `From<ChainstateError>` implementation.

Then, the caller of `validate()` could do:

```rust
proposal.validate(..).map_err(|reason| BlockValidateReject { block: proposal.block.clone(), reason_code: reason.reason_code, reason: reason.reason })
```

That way, this `validate()` function doesn't have to repeatedly perform that step.

Reply:
Done!